phys/module_physics_init.F

```diff
@@ -4653,17 +4655,52 @@ SUBROUTINE mp_init(RAINNC,SNOWNC,GRAUPELNC,config_flags,restart,warm_rain,
      END IF
 # endif
 #endif
-     CASE (NSSL_1MOMLFO)
-         CALL nssl_2mom_init(ims,ime, jms,jme, kms,kme,nssl_params,ipctmp=0,mixphase=0,ihvol=-1) ! no separate hail
-     CASE (NSSL_1MOM)
-         CALL nssl_2mom_init(ims,ime, jms,jme, kms,kme,nssl_params,ipctmp=0,mixphase=0,ihvol=0)
      CASE (NSSL_2MOM)
-         CALL nssl_2mom_init(ims,ime, jms,jme, kms,kme,nssl_params,ipctmp=5,mixphase=0,ihvol=1)
-     CASE (NSSL_2MOMG)
-         CALL nssl_2mom_init(ims,ime, jms,jme, kms,kme,nssl_params,ipctmp=5,mixphase=0,ihvol=-1) ! turn off hail
-     CASE (NSSL_2MOMCCN)
-         ccn_conc = nssl_cccn/1.225 ! set this to have correct boundary conditions
-         CALL nssl_2mom_init(ims,ime, jms,jme, kms,kme,nssl_params,ipctmp=5,mixphase=0,ihvol=1)
+
+#if (WRFPLUS != 1) & !defined( VAR4D )
+      IF ( config_flags%nssl_2moment_on == 0 ) THEN
+        nssl_ipconc = 0
+      ELSE
+        IF ( config_flags%nssl_3moment > 0 ) THEN
+          nssl_ipconc = 8
+        ELSE
+          nssl_ipconc = 5
+        ENDIF
+      ENDIF
+
+         IF ( config_flags % nssl_ccn_on > 0 ) THEN
+            ccn_conc = nssl_cccn/1.225 ! set this to have correct boundary conditions
+         ENDIF
+         CALL nssl_2mom_init(nssl_params=nssl_params,ipctmp=nssl_ipconc,mixphase=0, &
+           nssl_density_on=(config_flags%nssl_density_on > 0), &
+           nssl_hail_on=config_flags%nssl_hail_on > 0, &
+           nssl_ccn_on=(config_flags%nssl_ccn_on > 0), &
+           nssl_icdx=config_flags%nssl_icdx, &
+           nssl_icdxhl=config_flags%nssl_icdxhl,ccn_is_ccna=config_flags%nssl_ccn_is_ccna)
+
+!    CASE (NSSL_2MOM)
```

Comment:
@MicroTed Are the following cases now obsolete?

Reply:
Yes, I just neglected to remove these commented lines. The associated mp_physics values (17,19,21,22) are still checked in check_a_mundo and converted to 18 with appropriate flags (along with a warning).

Reply:
Okay. Let's go ahead and remove these then. If you think it would be helpful for posterity the comment could be reduced to something to the effect of `! Cases 17, 19, 21, 22 superseded by 18 <extra detail if needed>`

Reply:
Done